lib/Analysis/AxisInfo.cpp

```diff
@@ -511,8 +511,23 @@ class ExpandDimsOpAxisInfoVisitor final
     AxisInfo::DimVectorT contiguity = opInfo.getContiguity();
     AxisInfo::DimVectorT divisibility = opInfo.getDivisibility();
     AxisInfo::DimVectorT constancy = opInfo.getConstancy();
+    int64_t newDivisibility = 1;
+    if (opInfo.getConstantValue().has_value()) {
+      // The tensor is constant, same as ConstantOpAxisInfoVisitor
+      newDivisibility = highestPowOf2Divisor(opInfo.getConstantValue().value());
+    } else if (opInfo.getRank()) {
+      // Otherwise, calculate the GCD as the new divisibility
+      newDivisibility =
```

Comment:
Can you add a test case for this branch?

Reply:
I think it's probably still pessimistic though. 

If you have `[2, 4, 6, ..., 2n]` and expand it to 

```
[[2,
4,
6,
...,
2n]]
```

The divisibility should be updated to [2, 2] but actually we get [1, 2]. I remembered that @LyricZhao had a more complete solution.

Reply:
Yes, with this pull request, the divisibility will be updated to [2, 2] but not the previous [1, 2]. Please see the test case in test/Analysis/test-alignment.mlir, LINE 195, below.

Reply:
How about this [2, 3, 5, 6] (divisibility=2, contiguity=1) ->
```
[[2],
[3],
[5],
[6]]
```
Then it gets updated to divisibility = [2, 2] as well?

Reply:
I think the divisibility of [2,3,5,7] (or the GCD of 2,3,5,7) is not 2, but 1.

Reply:
Yeah, I think you're right

Reply:
> I think it's probably still pessimistic though.
> 
> If you have `[2, 4, 6, ..., 2n]` and expand it to
> 
> ```
> [[2,
> 4,
> 6,
> ...,
> 2n]]
> ```
> 
> The divisibility should be updated to [2, 2] but actually we get [1, 2]. I remembered that @LyricZhao had a more complete solution.

The solution in previous Slack discussion is to make a new field `total_divisibility` (which is the GCD of all elements) and make the current concept as `leading_divisibility`.

Reply:
Yes, splitting the divisibility into 2 different fields is a choice, and need more code. The current implementation, merging the two into one divisibility will need less change to the current code, but maybe confusing.

Reply:
Do I need to split it into the 2 divisibilities? Or stick to the current 1 divisibility?